Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php

```diff
@@ -148,7 +148,7 @@ private function handleCreateWorkspace(
                     $command->newContentStreamId,
                 )
             ),
-            ExpectedVersion::ANY()
```

Comment:
this slight change has a big impact as now after deleting a workspace a new one with the exact same name cannot be recreated.

And because `\Neos\Neos\Domain\Service\WorkspaceService::getUniqueWorkspaceName` only prevents naming conflict with existing workspaces that are projected attempting to create a workspace with the same title via Ui fails.

steps:
1. create workspace 'MyTest'
2. delete workspace
3. repeat

<img width="442" alt="image" src="https://github.com/user-attachments/assets/1129a827-872d-4479-b520-bdce64276850" />


The error is as expected:

> Workspace could not be created.
> Expected version: -1 [no stream], actual version: 1

That means imo that `getUniqueWorkspaceName` needs to handle this case.

Reply:
very good catch

Reply:
So this seems to hve no test then?

Reply:
jup the the core's `DeleteWorkspace` is fully untested - i have that somewhere in my todo list and stumble upon that every once and then. Also the cores `CreateWorkspace` is only tested implicit but not for such cases.

Reply:
As discussed: I reverted this part of the change.
To properly enforce hard constraints for workspace mutations, we'll need to project the version to the workspace read model. I'll try that in a follow-up, but this does not have to be changed for 9.0

Reply:
Ill try to push a simple test that tests exactly that create delete create works!